test/idp/src/main/kotlin/org/codice/compliance/web/sso/PostLoginTest.kt

```diff
@@ -56,21 +79,34 @@ class PostLoginTest : StringSpec() {
             response.statusCode shouldBe HTTP_OK
             val idpResponse = getServiceProvider(IdpResponder::class)
                     .getIdpPostResponse(response).decorate()
-
             idpResponse.bindingVerifier().verify()
 
             val responseDom = idpResponse.responseDom
-
             CoreVerifier(responseDom).verify()
-
-            ResponseProtocolVerifier(responseDom, TestCommon.ID).verify()
-
-            SingleSignOnProfileVerifier(responseDom).verify()
+            ResponseProtocolVerifier(responseDom, TestCommon.ID, acsUrl[SamlProtocol.Binding.HTTP_POST]).verify()
+            SingleSignOnProfileVerifier(responseDom, acsUrl[SamlProtocol.Binding.HTTP_POST]).verify()
         }
 
         "POST AuthnRequest With Relay State Test" {
-            val authnRequest = generateAndRetrieveAuthnRequest()
-            val encodedRequest = Encoder.encodePostMessage(authnRequest, EXAMPLE_RELAY_STATE)
+            val authnRequest = AuthnRequestBuilder().buildObject().apply {
```

Comment:
✏️ This block and the ones on line 52 and 135 are either identical or nearly so?

Can we extract to a setup method or extract a function to do the creation?